src/com/TTT/TurnUI.java

```diff
@@ -23,17 +23,34 @@ public void takeTurn(Board board, Game game) {
         String winningMarker = game.getPlayerMarker(game.currentPlayer());
 
         while(!validSelection(selectedSpace) || !game.selectionOnBoard(parseSelection(selectedSpace)) || board.isOccupied(parseSelection(selectedSpace))) {
-            selectedSpace = getPlayerInput(board, "Invalid input. " + inputPrompt);
+            selectedSpace = getPlayerInput(board,  invalidReasonText(board, game, selectedSpace) + inputPrompt);
         }
 
         game.doTurn(parseSelection(selectedSpace));
 
         if(game.isOver() == GameState.Win) {
-            out.print("Player " + winningMarker + " is the winner.");
+            printer.printBoard(board);
+            out.print("Player " + winningMarker + " is the winner.\n");
         }
 
         if (game.isOver() == GameState.Tie){
-            out.print("This game is a tie.");
+            printer.printBoard(board);
+            out.print("This game is a tie.\n");
+        }
+    }
+
+    private String invalidReasonText(Board board, Game game, String selectedSpace) {
+        if (!validSelection(selectedSpace)) {
```

Comment:
You don't have to action now, but could be worth having a look at the `Strategy` pattern to see if it would help here. Sometimes it can be helpful when you have a lot of similar 'if/else' statements.